File: array/566.py
class Solution:
    def matrixReshape(self, nums, r, c):
        """
        :type nums: List[List[int]]
        :type r: int
        :type c: int
        :rtype: List[List[int]]
        """
        if not nums:
            return []
        
        col = len(nums[0])
        row = len(nums)
        
        if not row*col == r*c:
            return nums
        
        ret = [[0]*c for i in range(r)]
        temp = []
        for i in range(row):
            for j in range(col):
                temp.append(nums[i][j])
        
        for i in range(r):
            for j in range(c):
                ret[i][j] = temp.pop(0)
        return ret
                        
                    
                
        

File: array/566.cc
class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>>& mat, int r, int c) {
        if (mat.size() * mat[0].size() != r * c)
            return mat;
        std::vector<std::vector<int>> ret;
        std::vector<int> cur;
        int index_r = 0, index_c = 0;
        for (const auto &line : mat) {
            for (const auto &num : line) {
                if (index_c < c) {
                    cur.emplace_back(num);
                    ++index_c;
                } else {
                    std::vector<int> temp;
                    cur.swap(temp);
                    ret.emplace_back(temp);
                    cur.emplace_back(num);
                    index_c = 1;   
                }
            }
        } 
        ret.emplace_back(cur);
        return ret;
    }
};
